app/controllers/FileUploadController.scala

```diff
@@ -94,6 +73,80 @@ class FileUploadController @Inject()(fileUploadConnector: FileUploadConnector,
+                    val error = extractErrorReason(session.uploadResponse)
+                    if (error == Messages("upload.failed.error")) {
+                      sessionService.cacheUploadResponse(UploadResponse("", None)).map {
+                        case Some(_) =>
+                          Redirect(routes.ErrorController.renderProblemUploadingFilePage())
+                        case _ =>
+                          Logger.error(s"[FileUploadController][post] failed to obtain a session for userId ($userId)")
+                          Redirect(routes.ErrorController.renderGlobalErrorPage())
+                      }
+                    }
+                    else {
+                      sessionService.resetCacheUploadResponse()
+                      Future.successful(Ok(views.html.file_upload(form, error)))
+                    }
+                  case _ =>
+                    Logger.error(s"[FileUploadController][post] failed to obtain a form url using existing envelope " +
+                      s"for userId ($userId)")
+                    Future.successful(Redirect(routes.ErrorController.renderGlobalErrorPage()))
+                }
+              case _ =>
+                createFileUploadUrl(None, userId)(request, hc).flatMap {
+                  case Some(url) =>
+                    Logger.info(s"[FileUploadController][post] stored new envelope id successfully for userId ($userId)")
+                    http.POSTForm(url, request.body.asMultipartFormData.get.dataParts, request.headers.headers).map{ _ =>
+                      Redirect(routes.FileUploadController.uploadSuccess())
+                    } recover {
+                      case err =>
+                        Logger.info(s"[FileUploadController][post] file upload failed", err)
+                        Redirect(routes.FileUploadController.uploadError())
+                    }
+                  case _ =>
+                    Logger.error(s"[FileUploadController][post] failed to obtain a form url using new envelope for userId ($userId)")
+                    Future.successful(Redirect(routes.ErrorController.renderGlobalErrorPage()))
+                }
+            }.recover {
+              case e: Throwable =>
+                Logger.error(s"[FileUploadController][post] failed to fetch ras session for userId ($userId) - $e")
+                Redirect(routes.ErrorController.renderProblemUploadingFilePage())
+            }
+        }
+      case Left(res)
+      =>
+        Logger.warn("[FileUploadController][post] user Not authorised")
+        res
+    }
+  }
+
+  def getFile(request: Request[AnyContent]): FilePart[Files.TemporaryFile]  = {
+    val requestBody = request.body.asMultipartFormData.getOrElse("")
+    requestBody match {
+      case body:MultipartFormData[Files.TemporaryFile] => body.file("file").get
+      case _ => FilePart("","",Some("text"), TemporaryFile(""))
+    }
+  }
```

Comment:
We still have a .get on an optional value here that can be refactored out. Maybe it can be replaced with something like this?

` def getFile(request: Request[AnyContent]): FilePart[Files.TemporaryFile]  = {

    val fallBackFilePart = FilePart("","",Some("text"), TemporaryFile(""))
    request.body.asMultipartFormData match {
      case Some(body) => body.file("file").getOrElse(fallBackFilePart)
      case _ => fallBackFilePart
    }
  }
`